src/cmd/run.js

```diff
@@ -42,16 +44,28 @@ export function defaultWatcherCreator(
   {
     addonId, client, sourceDir, artifactsDir,
     onSourceChange = defaultSourceWatcher,
+    desktopNotifications = notifier,
   }: WatcherCreatorParams
  ): Watchpack {
   return onSourceChange({
     sourceDir,
     artifactsDir,
     onChange: () => {
       log.debug(`Reloading add-on ID ${addonId}`);
+
       return client.reloadAddon(addonId)
+        .then(() => {
+          desktopNotifications({
```

Comment:
@rpl @kumar303 i've removed notification on reload and corresponding test for now, once the code is reviewed and decision is made on how to better handle it, we can continue working on it.